mpi.h-v3-constified

```diff
@@ -0,0 +1,177 @@
+# A list of functions to intercept via PMPI. This list is meant to be used with MPI version 3. Many functions are presently missing or commented out, in need of update.
+
+int MPI_Abort(MPI_Comm comm, int errorcode);
+int MPI_Address(void *location, MPI_Aint *address);
+int MPI_Allgather(const void *sendbuf, int sendcount, MPI_Datatype sendtype, void *recvbuf, int recvcount, MPI_Datatype recvtype, MPI_Comm comm);
+int MPI_Allgatherv(const void *sendbuf, int sendcount, MPI_Datatype sendtype, void *recvbuf, const int *recvcounts, const int *displs, MPI_Datatype recvtype, MPI_Comm comm);
+int MPI_Allreduce(const void *sendbuf, void *recvbuf, int count, MPI_Datatype datatype, MPI_Op op, MPI_Comm comm);
+int MPI_Alltoall(const void *sendbuf, int sendcount, MPI_Datatype sendtype, void *recvbuf, int recvcount, MPI_Datatype recvtype, MPI_Comm comm);
+int MPI_Alltoallv(const void *sendbuf, const int *sendcounts, const int *sdispls, MPI_Datatype sendtype, void *recvbuf, const int *recvcounts, const int *rdispls, MPI_Datatype recvtype, MPI_Comm comm);
+int MPI_Alltoallw(const void *sendbuf, const int *sendcounts, const int *sdispls, const MPI_Datatype *sendtypes, void *recvbuf, const int *recvcounts, const int *rdispls, const MPI_Datatype *recvtypes, MPI_Comm comm);
+int MPI_Barrier(MPI_Comm comm);
+int MPI_Bcast(void *buffer, int count, MPI_Datatype datatype,      int root, MPI_Comm comm);
```

Comment:
Some of the spacing in arguments throughout this file is weird.

Reply:
I see it's just because it's a copy of the existing function listings, so it's fine for now, they can all be prettified in a separate commit later.